parser/is-binary.go

```diff
@@ -30,12 +31,25 @@ func HasBinary(data any) bool {
 				return true
 			}
 		}
+		return false
 	case map[string]any:
 		for _, v := range o {
 			if HasBinary(v) {
 				return true
 			}
 		}
+		return false
+	}
+	dv := reflect.ValueOf(data)
+	if dv.Kind() == reflect.Struct {
```

Comment:
Fair enough!  I wasn't sure how well updated this library was and whether you'd like the PR at all since it feels a little hacky to me, so wanted to get something basic out first. :D  Will add this...

Reply:
Well, given this feature, we can even do more, such as support for Array and Map.

Reply:
done!